modules/appmonetBidAdapter.js

```diff
@@ -0,0 +1,279 @@
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO],
+
+  isBidRequestValid(bid) {
+    return nullOrType(deepAccess(bid, 'params.endpoint', null), 'string') &&
+      nullOrType(deepAccess(bid, 'params.tagId', null), 'string') &&
+      nullOrType(deepAccess(bid, 'params.testMode', null), 'boolean');
+  },
+
+  buildRequests(bidRequests, bidderRequest) {
+    const loc = getLocation(bidderRequest);
+    const tagId = deepAccess(bidRequests[0], 'params.tagId', null);
+    const testMode = deepAccess(bidRequests[0], 'params.testMode', 0);
+
+    const payload = {
+      a: bidderRequest.auctionId,
+      B: 0,
+      b: loc.host,
+      tm: testMode,
+      V: '$prebid.version$',
+      i: (testMode && tagId != null) ? tagId : getID(loc),
+      l: {},
+      f: 0.01,
+      cv: VERSION,
+      st: 'prebid',
+      h: screen.height,
+      w: screen.width,
+      gs: deepAccess(bidderRequest, 'gdprConsent.gdprApplies', '0'),
+      gc: deepAccess(bidderRequest, 'gdprConsent.consentString', ''),
+      u: deepAccess(bidderRequest, 'refererInfo.canonicalUrl', loc.href),
+      do: loc.host,
+      re: deepAccess(bidderRequest, 'refererInfo.referer'),
+      usp: bidderRequest.uspConsent || '1---',
+      smt: 9,
+      d: '',
+      m: createBidMap(bidRequests),
+    };
+
+    return {
+      data: payload,
+      method: 'POST',
+      url: deepAccess(bidRequests[0], 'params.endpoint', DEFAULT_ENDPOINT),
+      withCredentials: true,
+    };
+  },
+
+  getUserSyncs(syncOptions, serverResponses) {
+    return (serverResponses || [])
+      .flatMap(({ body: response }) =>
+        response != null && response.p != null ? (response.p.hreq || []) : [])
+      .map((syncPixel) =>
+        ({
+          type: syncPixel.indexOf('__st=iframe') !== -1 ? 'iframe' : 'image',
+          url: syncPixel
+        })
+      ).filter(({
+        type
+      }) => syncOptions.iframeEnabled || type === 'image')
+  },
+
+  interpretResponse(serverResponse, request) {
+    // validate the body/response
+    const response = serverResponse.body;
+    if (response == null || typeof response === 'string') {
+      return [];
+    }
+
+    return Object.keys(response.r).flatMap((bidID) => {
+      const biddata = response.r[bidID];
+      return biddata.flatMap((siteBid) =>
+        siteBid.b.map((bid) => {
+          const mediaType = getMediaType(bid);
+          const ad = mediaType === BANNER ? decorateADM(bid) : decorateVideoADM(bid);
+          if (ad == null) {
+            return null;
+          }
+
+          const size = resolveSize(bid, request.data, bidID);
+
+          return ({
+            requestId: bidID,
+            cpm: bid.price,
+            width: size[0],
+            height: size[1],
+            creativeId: bid.crid,
+            currency: 'USD',
+            netRevenue: true,
+            [mediaType === VIDEO ? 'vastXml' : 'ad']: ad,
+            meta: {
+              advertiserDomains: bid.adomain,
```

Comment:
Hi @patmmccann ! We set `meta.advertiserDomains` here—is this what you mean or are you talking about a different place?

Reply:
awesome, ty!